### ISSUE TITLE
Add alternative style of talks like Telegram Web

### ISSUE THREAD

**Issue by shawnshaddock** [telegramdesktop/tdesktop#321]
The mobile style left to right message bubbles are not appropriate for the desktop due to the size of the screens.

It makes it much harder to follow a conversation because of the amount of space (left to right) between the messages. Please keep the messages left aligned in the desktop implementation.

For a comparison see the web vs the desktop implementation. The web one is much easier to follow.

Web
![telegramweb](https://cloud.githubusercontent.com/assets/6936864/5428084/4474d2d6-8384-11e4-950a-043e250dc198.png)

Desktop
![telegramdesktop](https://cloud.githubusercontent.com/assets/6936864/5428086/48f98298-8384-11e4-98b0-1ff172579a10.png)

**Comment by d87** [comment]
I agree, it's really annoying on desktop

**Comment by haevalencia** [comment]
+1000 new design is good, but web telegram works better

**Comment by selfus** [comment]
Are we still working on this, or do we have to eat whatever the devs think is best, when clearly it isn't?

**Comment by stek29** [comment]
@selfus you don't like it -- you don't eat it.
It's open source app, go and cook something suitable for you.

**Comment by Storm-Engineer** [comment]
@Aokromes I'm not sure that the issues you recently closed are really duplicates of this one. What we were asking is to have manual toggle for a feature that already exists, instead of (/in addition to) it being toggled arbitrarily based on window width.

**Comment by Aokromes** [comment]
well, addition of can be permanent toggle of the feature :P

**Comment by Storm-Engineer** [comment]
So, do you think it's something we can expect in the near future, considering that lot of people would like it and it shouldn't be too hard to do?

I wish I could just offer to contribute but I I'm not a GUI programmer.

**Comment by Aokromes** [comment]
if someone makes pull requests for alternative styles and they are good (or as simply as "permanent left aligned text" for sure preston will merge them.

**Comment by Aokromes** [comment]
with the use of https://woboq.com/blog/qdockwidget-changes-in-56.html users can customize how telegram looks :)

**Comment by Storm-Engineer** [comment]
So, all tickets about this issue are closed but nobody actually attempted to do anything about the issue, it seems?

**Comment by Storm-Engineer** [comment]
So, is someone working on this? If no, how could we get some attention on it and find someone who could and is willing to work on it?

Sadly I don't have the knowledge to do it.

**Comment by Storm-Engineer** [comment]
Damnit stupid bot, don't close issues that still exist!!!

**Comment by ilya-fedin** [comment]
> Damnit stupid bot, don't close issues that still exist!!!

post a message when bot asks you if you don't want an issue to be closed

**Comment by Storm-Engineer** [comment]
I'm sorry, I didn't actually see that email until today. And thank you for re-opening.

Although it doesn't seem like anyone wants to implement this and sadly I don't have the knowledge.

**Comment by ilya-fedin** [comment]
> Although it doesn't seem like anyone wants to implement this and sadly I don't have the knowledge.

Even if someone will implement this, only Telegram designers have permissions to change design of official apps and PR will be just rejected

**Comment by Storm-Engineer** [comment]
> > Although it doesn't seem like anyone wants to implement this and sadly I don't have the knowledge.
> 
> Even if someone will implement this, only Telegram designers have permissions to change design of official apps and PR will be just rejected

Why? I think it is a reasonable request, and it doesn't go against the existing design, since we are simply asking for an existing feature to be manual-toggle instead of automatic, or to be consistent with how the web version works.

**Comment by ilya-fedin** [comment]
> simply asking for an existing feature

What do you mean by "existing feature"? There's no second hidden design.

**Comment by Storm-Engineer** [comment]
> What do you mean by "existing feature"? There's no second hidden design.
The ability to have all chat bubbles on the left, instead of having your own on the right. This is clearly said in the ticket. This is an existing feature, since in an ultra-wide window size it happens automatically, and on the web version this is the default.

But on the desktop version, your own messages will be on the right, unless the window is made ultra-wide. This is demonstrated in the screenshots in this very ticket. It is extremely inconvenient and annoying having to constantly jump left-right-left-right to be able to read the conversation. Plus, some people like me simply hate the left-right design altogether and just want everything on the left. Desktops are not phones and forcing a design meant for phones on a desktop app is a bad approach.

This ticket, and several others that were closed as a duplicate of this, including mine, are asking for a simple checkbox, to make your own messages always stay on the left, regardless of window width, instead of the current behavior that is both arbitrary and inconsistent with how the web version works.

**Comment by RememberTheAir** [comment]
@Storm-Engineer I agree with what everything you said - I just want to point out though that the web version has long been discontinued (there's a new one wip I guess), so its design implementation shouldn't be used as a reference

**Comment by ilya-fedin** [comment]
> This is clearly said in the ticket.

Not so clearly, I read it as completely redesign without bubbles

**Comment by ilya-fedin** [comment]
The problem here is that @john-preston is against adding any new options, he wants to have as less options as possible.

**Comment by Storm-Engineer** [comment]
> The problem here is that @john-preston is against adding any new options, he wants to have as less options as possible.

Even at the cost of hurting users' needs? If you look at the _dozens_ of tickets requesting this, you will see that this is a problem for lot of people.

But if they insist on keeping the settings UI limited at all cost, just make this possible via a config file. I'm pretty sure those who really need this feature wouldn't mind having to edit a text file to enable it. Win-win?

**Comment by LukasWallisch** [comment]
Maybe it would fit perfekt into the "Theme" Menu... that single additional option would make no difference...

**Comment by john-preston** [comment]
Problem with that is not the option, but the amount of coding required to add support for that.

**Comment by ilya-fedin** [comment]
@john-preston looks like @Aokromes over-optimized issues and half of people here just want https://github.com/telegramdesktop/tdesktop/issues/5880

**Comment by Storm-Engineer** [comment]
> @john-preston looks like @Aokromes over-optimized issues and half of people here just want #5880

Sorry for the [very] late reply, but YES. Exactly.

**Comment by ilya-fedin** [comment]
> Sorry for the [very] late reply, but YES. Exactly.

It's unlikely old issues would be reopened, but you can create a new one
